superset/assets/src/SqlLab/components/SouthPane.jsx

```diff
@@ -39,6 +42,12 @@ class SouthPane extends React.PureComponent {
       latestQuery = props.editorQueries[props.editorQueries.length - 1];
```

Comment:
if it is in offline state, you don't need to run these logic, you just render offline and exit early, right? so you can move offline check before this line.